assets/src/blocks/Media/MediaElementVideo.js

```diff
@@ -0,0 +1,60 @@
+import { useEffect } from '@wordpress/element';
+import { useScript } from '../../components/useScript/useScript';
+import { useStyleSheet } from '../../components/useStyleSheet/useStyleSheet';
+
+export const MediaElementVideo = ({ videoURL, videoPoster }) => {
+
+	const [stylesLoaded, stylesError] = useStyleSheet(
+    `https://cdn.jsdelivr.net/npm/mediaelement@4.2.16/build/mediaelementplayer-legacy.min.css`
+	);
+
+	const [scriptLoaded, scriptError] = useScript(
+		`https://cdn.jsdelivr.net/npm/mediaelement@4.2.16/build/mediaelement-and-player.min.js`
+	);
+
+	const [vimeoRendererLoaded, vimeoRendererError] = useScript(
+		`https://cdn.jsdelivr.net/npm/mediaelement@4.2.16/build/renderers/vimeo.min.js`
+	);
+
+  const setupMediaElementJS = function() {
+		const meJSNodes = document.querySelectorAll( '.mejs-video-block' );
+
+		meJSNodes.forEach(node => {
```

Comment:
What is the behavior here if a page has many video blocks? Does each render also setup the player for all other blocks n times?

Reply:
Oops, forgot to update this, it should use a ref to the node as the Timeline does.

Reply:
So there was a problem with using the `useScript` and `useStylesheet` hooks: we can't use hooks in the `save()` method. So I extracted the part of the code that actually creates the script/link elements and adds them to the DOM into two functions that can be used inside the hooks or without them.

Reply:
Maybe we can use `hydrate` on the frontend and still be able to use these hooks?

Reply:
Wouldn't `hydrate` miss the point of making the block "static" and resilient to change? Since these are not server side rendered by React but saved statically at editing time into the database, any change in the component would trigger a hydration warning.

Reply:
That is, any change that causes the resulting markup to be different.

Reply:
It would indeed make the block harder to change, but if we know the markup will rarely change that might be less of an issue. It would have the advantage of not having some separate logic somewhere else that needs to be called for the block to work.